### DISCUSSION TITLE
Could ValidateResult / Message be also a ReactNode?

### DISCUSSION THREAD

**Discussion by dqbd** [react-hook-form/react-hook-form#4289] [Q&A] [answered]
Hello! Thanks for the great library 🙏 Just have a small question:

I currently have an usecase, where I needed to return a ReactNode as a validation error message and was wondering, whether it would be possible to extend the type of `Message` to also include ReactNode?

Example usage:
```
<input
   validate={{ match: () => <span>Hello</span> }}
/>
```

**Comment by bluebill1049** [comment, marked as answer]
currently, we only support string for `TS` (not a problem with JS), as we have limitations on type support for both string and JSX.

**Reply by dqbd** [reply]
Right, will ignore the TS error for now, would you recommend any other patterns for rendering different JSX based on error messages? I think returning some custom string IDs should work fine?

**Reply by bluebill1049** [reply]
you can use the `type` form the error message to render different messages.

**Reply by dqbd** [reply]
Right, makes sense, thanks 😃

**Reply by alpacachen** [reply]
hi, will you plan to support reactNode type message in TS?
